src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/listeners/court/libra/LibraCourtCaseListenerIntTest.kt

```diff
@@ -110,47 +111,45 @@ class LibraCourtCaseListenerIntTest : MessagingMultiNodeTestBase() {
   }
 
   @Test
-  fun `should process send correct telemetry for candidate search`() {
-    personRepository.saveAndFlush(
-      PersonEntity.from(
-        Person(
-          firstName = "Jane",
-          lastName = "Smith",
-          addresses = listOf(Address(postcode = "LS1 1AB")),
-          sourceSystemType = SourceSystemType.LIBRA,
-        ),
+  fun `should send correct telemetry for candidate search when one match found`() {
+    val firstName = UUID.randomUUID().toString()
+    val firstMessageId = publishHMCTSMessage(libraHearing(firstName = "Steve", lastName = "Micheal", postcode = "RF14 5DG"), LIBRA_COURT_CASE)
+    val secondMessageId = publishHMCTSMessage(libraHearing(firstName = firstName, lastName = "Smith", postcode = "LS1 1AB"), LIBRA_COURT_CASE)
+    checkTelemetry(
+      CPR_CANDIDATE_RECORD_SEARCH,
+      mapOf(
+        "SOURCE_SYSTEM" to "LIBRA",
+        "EVENT_TYPE" to LIBRA_COURT_CASE.name,
+        "SEARCH_VERSION" to PersonSpecification.SEARCH_VERSION,
+        "MESSAGE_ID" to firstMessageId,
       ),
     )
-    personRepository.saveAndFlush(
-      PersonEntity.from(
-        Person(
-          firstName = "Steve",
-          lastName = "Micheal",
-          addresses = listOf(Address(postcode = "RF14 5DG")),
-          sourceSystemType = SourceSystemType.LIBRA,
-        ),
+    checkTelemetry(
+      CPR_CANDIDATE_RECORD_SEARCH,
+      mapOf(
+        "SOURCE_SYSTEM" to "LIBRA",
+        "EVENT_TYPE" to LIBRA_COURT_CASE.name,
+        "SEARCH_VERSION" to PersonSpecification.SEARCH_VERSION,
+        "MESSAGE_ID" to secondMessageId,
       ),
     )
-
-    await untilAsserted { assertThat(personRepository.findAll().size).isEqualTo(2) }
-
-    val messageId = publishHMCTSMessage(
+    val thirdMessageId = publishHMCTSMessage(
       libraHearing(
-        firstName = "Jane",
+        firstName = firstName,
         lastName = "Smythe",
         postcode = "LS1 1AB",
       ),
       LIBRA_COURT_CASE,
     )
 
     checkTelemetry(
-      TelemetryEventType.CPR_CANDIDATE_RECORD_SEARCH,
+      CPR_CANDIDATE_RECORD_SEARCH,
       mapOf(
         "SOURCE_SYSTEM" to "LIBRA",
         "EVENT_TYPE" to LIBRA_COURT_CASE.name,
         "RECORD_COUNT" to "1",
         "SEARCH_VERSION" to PersonSpecification.SEARCH_VERSION,
-        "MESSAGE_ID" to messageId,
+        "MESSAGE_ID" to thirdMessageId,
       ),
```

Comment:
This test is failing because the number of records found is more like 270 than 1. Using a GUID for the first name should (I think) ensure that the soundex can only find one record. Food for thought

Reply:
Not if the PNC/CRO always matches of course, and we use the same PNC/CRO in all tests